src/js/helpers/useToast.test.ts

```diff
@@ -0,0 +1,122 @@
+import selectorsMap from '@constants/selectors-map';
+import useToast from '@helpers/useToast';
+import { 
+  containerComponent, btnCloseComponent, testMessage, testOptions, defaultType, validTemplate, invalidTemplate 
+} from '@constants/mocks/useToast-data';
+
+describe('useToast', () => {
+  const expectedOptions = {
+    getType(hasType: boolean) {
+
+      return hasType ? `.bg-${testOptions.type}` : `.bg-${defaultType}`;
+    },
+    getClassList() {
+      const customClassList = testOptions.classlist;
+      let expectedClassList = '';
+
+      if (customClassList) {
+        customClassList.trim().split(' ').forEach((value) => {
+          
+          if (value) {
+            expectedClassList = expectedClassList.concat(' ', value);
+          }
+        });
+
+        return expectedClassList.trim();
+      }
+    },
+    getBtnClose() {
+      if (testOptions.autohide !== undefined && testOptions.autohide === false) {
+
+        return `.${btnCloseComponent}`;
+      }
+    }
+  }
+
+  describe('with valid template', () => {
```

Comment:
```suggestion
  describe('with container and toast template existing in the DOM ', () => {
```

I would make it a bit more verbose